debezium-operator-core/src/main/java/io/debezium/operator/core/dependent/RoleDependent.java

```diff
@@ -23,16 +27,31 @@ public RoleDependent() {
 
     @Override
     protected Role desired(DebeziumServer primary, Context<DebeziumServer> context) {
+
         return new RoleBuilder()
                 .withMetadata(new ObjectMetaBuilder()
                         .withName(ROLE_NAME.formatted(primary.getMetadata().getName()))
                         .withNamespace(primary.getMetadata().getNamespace())
                         .build())
                 .withRules(new PolicyRuleBuilder()
                         .withApiGroups("")
-                        .withResources("secrets", "configmaps")
+                        .withResources("secrets")
                         .withVerbs("get", "list", "watch")
-                        .build())
+                        .build(),
+                        new PolicyRuleBuilder()
+                                .withApiGroups("")
+                                .withResources("configmaps")
+                                .withVerbs(getConfigMapPermissions(primary))
+                                .build())
                 .build();
     }
+
+    private static List<String> getConfigMapPermissions(DebeziumServer primary) {
+
+        List<String> configMapPermissions = new ArrayList<>(List.of("get", "list", "watch"));
+        if (primary.getSpec().getSource().getOffset().getActiveStore() instanceof ConfigMapOffsetStore) {
+            configMapPermissions.addAll(List.of("create", "update", "patch"));
```

Comment:
In case of operator, I would actually leave the creation of the CM on the operator. Thus create should not be required